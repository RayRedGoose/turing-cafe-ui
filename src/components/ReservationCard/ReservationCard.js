import React from 'react'
import './ReservationCard.scss'

const ReservationCard = ({info, removeReserv}) => {
  const {id, name, date, time, number} = info

  const deleteCard = (event) => {
    event.preventDefault()
    removeReserv(id)
  }

  return (
    <section id={id} className="reserv-card">
      <h1>{name}</h1>
      <p>{date}</p>
      <p>{time}</p>
      <p>Number of guests: {number}</p>
      <button onClick={deleteCard}>cancel</button>
    </section>
  )
}

export default ReservationCard
