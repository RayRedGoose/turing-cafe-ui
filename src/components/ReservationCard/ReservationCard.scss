.reserv-card {
  background-color: #fff;
  border: 5px solid #088081;
  border-radius: 10px;
  color: #088081;
  margin: 0 20px 20px;
  padding: 20px;
  min-width: 13%;
  button {
    width: 100%;
  }
}
