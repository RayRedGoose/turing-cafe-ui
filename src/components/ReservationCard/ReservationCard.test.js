import React from 'react'
import { shallow } from 'enzyme'
import ReservationCard from './ReservationCard'

describe("ReservationCard", () => {
  let card, mockProps
  const removeReserv = jest.fn()

  beforeEach(() => {
    mockProps = {
      id: 1,
      name: 'Ray',
      date: '12/12',
      time:'5:00',
      number: '5'
    }

    card = shallow(
      <ReservationCard
        key={mockProps.id}
        info={mockProps}
        removeReserv={removeReserv} />
    )
  })

  it('renders with all data passed in correctlly', () => {
    expect(card).toMatchSnapshot()
  })

  it('should call removeReserv prop when button is clicked', () => {
    card.find('button').simulate(
      'click',
      {preventDefault: jest.fn()}
    )

    expect(removeReserv).toHaveBeenCalled()
  })
})
