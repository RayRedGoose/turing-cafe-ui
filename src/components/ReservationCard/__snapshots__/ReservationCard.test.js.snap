// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationCard renders with all data passed in correctlly 1`] = `
<section
  className="reserv-card"
  id={1}
>
  <h1>
    Ray
  </h1>
  <p>
    12/12
  </p>
  <p>
    5:00
  </p>
  <p>
    Number of guests: 
    5
  </p>
  <button
    onClick={[Function]}
  >
    cancel
  </button>
</section>
`;
